src/main/java/uk/gov/hmcts/ccd/domain/service/caseaccess/CaseAccessOperation.java

```diff
@@ -84,25 +92,118 @@ public void updateUserAccess(CaseDetails caseDetails, CaseUser caseUser) {
     }
 
     @Transactional
-    public void addCaseUserRoles(List<CaseAssignedUserRole> caseUserRoles) {
+    public void addCaseUserRoles(List<CaseAssignedUserRoleRequest> caseUserRoles) {
+
+        Map<Long, List<CaseAssignedUserRoleRequest>> cauRolesByCaseId = getMapOfCaseAssignedUserRolesByCaseId(caseUserRoles);
+
+        Map<String, Map<String, Object>> newUserCounts = getNewUserCountByCaseAndOrganisation(cauRolesByCaseId);
+
+        cauRolesByCaseId.forEach((caseId, roles) ->
```

Comment:
in the `value` of an entry, the list is not that of roles. perhaps a name like requestedAssignments could read